### ISSUE TITLE
Docker daemon gets stuck on startup

### ISSUE THREAD

**Issue by kekbur** [docker/docker#27627]
**Description**

After upgrading to the latest version, Docker engine gets stuck on startup. The following is written in /var/log/daemon.log.

```
Oct 21 11:38:20 dev systemd[1]: Starting Docker Socket for the API.
Oct 21 11:38:20 dev systemd[1]: Listening on Docker Socket for the API.
Oct 21 11:38:20 dev systemd[1]: Starting Docker Application Container Engine...
Oct 21 11:38:20 dev dockerd[2044]: time="2016-10-21T11:38:20.792060229+03:00" level=info msg="libcontainerd: new containerd process, pid: 2048"
Oct 21 11:38:20 dev dockerd[2044]: time="2016-10-21T11:38:20.800498702+03:00" level=fatal msg="json: cannot unmarshal number -1 into Go value of type uint32"
Oct 21 11:38:26 dev dockerd[2044]: time="2016-10-21T11:38:26.794982037+03:00" level=info msg="libcontainerd: new containerd process, pid: 2056"
Oct 21 11:38:26 dev dockerd[2044]: time="2016-10-21T11:38:26.802289058+03:00" level=fatal msg="json: cannot unmarshal number -1 into Go value of type uint32"
```

**Steps to reproduce the issue:**
1. sudo service docker start
2.
3.

**Describe the results you received:**
dockerd process starts, produces the error messages in the log, does not exit, does not accept connections.

**Describe the results you expected:**
Docker engine starts.

**Additional information you deem important (e.g. issue happens only occasionally):**
Upgrade succeeded on another machine with identical setup, but different containers.

**Output of `docker version`:**

```
Client:
 Version:      1.12.2
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   bb80604
 Built:        Tue Oct 11 17:53:36 2016
 OS/Arch:      linux/amd64
Cannot connect to the Docker daemon. Is the docker daemon running on this host?
```

**Output of `docker info`:**

```
Cannot connect to the Docker daemon. Is the docker daemon running on this host?
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
```
root@dev:~# uname -a
Linux dev 3.16.0-4-amd64 #1 SMP Debian 3.16.7-ckt20-1+deb8u4 (2016-02-29) x86_64 GNU/Linux
```

```
root@dev:~# cat /etc/debian_version 
8.6
```

```
root@dev:~# cat /etc/apt/sources.list.d/docker.list 
deb https://apt.dockerproject.org/repo debian-wheezy main
```

**Comment by thaJeztah** [comment]
What version were you running before upgrading? IIRC, there was a change during one of the betas/release candidates that could cause this

**Comment by kekbur** [comment]
According to APT logs the upgrade was as follows: ```docker-engine:amd64 (1.12.1-0~wheezy, 1.12.2-0~wheezy)```

So the previous version was ```1.12.1-0~wheezy```.

**Comment by mlaventure** [comment]
@kekbur can you restart one more time with debug log activated?

**Comment by crosbymichael** [comment]
This has been fixed by #27693

**Comment by chulkilee** [comment]
It happened on centos7, from `docker-engine-1.12.1-1.el7.centos.x86_64` to `docker-engine-1.12.2-1.el7.centos.x86_64`.

I fixed it by moving `/var/run/docker/libcontainerd/containerd/events.log` somewhere.
